### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by JHops881**
decoupled the client and server. Initial porotype of client-server communications. Other changes. Highly likely to change, in fact, another overhaul is planned to changed to ECS. At least this can be in the books to prove that it is possible.

### CODE REVIEW

File: server/src/player_projectile.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        }
+    }
+
+    /// Call this every fixed game update.
+    pub fn update(&mut self, delta_time: f32) {
+        if self.expiriation_time > system_time() {
+            self.translate(delta_time)
+        } else {
+            // Ideally, we would drop the damn thing right here. But it is
+            // a tough cookie. Can't drop by reference at all. We need to
```
Comment (anultravioletaurora) [inline server/src/player_projectile.rs:73]: 🍪💪

File: server/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    // game loop crap
+    // https://fulmanski.pl/zajecia/tippgk/zajecia_20162017/wyklad_cwiczenia_moje/game_loop_and_time.pdf
+    let mut real_delta_time: f64;
+    let mut last_update_time: f64 = system_time();
+    let target_time_frame: f64 = 15.625; // 64 tick
+    let mut accumulator: f64 = 0.0;
+
+    // While game is running...
+    loop {
```
Comment (anultravioletaurora) [inline server/src/main.rs:68]: One thing you might consider is pulling out some of the code in here and making methods for the things that occur, such as taking the loop code and making that a method. Nothing wrong with what you have here, just makes the code a little easier to read

**Review by anultravioletaurora — APPROVED** [review]
LGTM! One thing I'll call out is the loop in the main method could be broken up a little bit, but again I think it all looks good.

I'm not sure if Rust let's you have like a properties file with configuration in it, but some hardcoded variable values could then be placed in there so they're all in one spot. Makes it a little easier to change those values instead of hunting in the code to see where they are set.

Aside from those nitpicks, I think this looks good. You've got great comments in there and the code seems pretty self documented too. Nice work!

File: server/src/net.rs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        // each client's main character.
+        for player in game_world.player_characters.iter() {
+            if player.get_uuid() == *id {
+                let player_data: PlayerData = PlayerData {
+                    x: player.get_position().x,
+                    y: player.get_position().y,
+                    id: player.get_uuid(),
+                    name: player.get_name(),
+                    speed_stat: player.get_speed_stat(),
+                    health_stat: player.get_health_stat(),
```
Comment (lavahoppers) [inline server/src/net.rs:73]: i really love cats

**Pull request merged**